test/modules/core/core/animation-loop.spec.js

```diff
@@ -71,3 +71,78 @@ test('WebGL#AnimationLoop redraw', t => {
     }
   }).start();
 });
+
+test('WebGL#AnimationLoop two start()s should only run one loop', t => {
+  if (typeof document === 'undefined') {
+    t.comment('browser-only test');
+    t.end();
+    return;
+  }
+
+  const {gl} = fixture;
+  let initializeCalled = 0;
+
+  const animationLoop = new AnimationLoop({
+    gl,
+    onInitialize: () => {
+      initializeCalled++;
+    },
+    onRender: () => {
+      animationLoop.stop();
+      // FIXME: this is not directly test that only one loop has been started
+      t.is(initializeCalled, 1, 'onInitialize called');
+      t.end();
+    }
+  });
+  animationLoop.start();
+  animationLoop.start();
+});
+
+test('WebGL#AnimationLoop start followed immediately by stop() should stop', t => {
+  if (typeof document === 'undefined') {
+    t.comment('browser-only test');
+    t.end();
+    return;
+  }
+
+  const {gl} = fixture;
+  let initializeCalled = 0;
+
+  const animationLoop = new AnimationLoop({
+    gl,
+    onInitialize: () => {
+      initializeCalled++;
+    }
+  });
+  animationLoop.start();
+  animationLoop.stop();
+  setTimeout(() => {
+    t.is(initializeCalled, 0, 'onInitialize called');
+    t.end();
+  }, 100);
+});
+
+test('WebGL#AnimationLoop a start/stop/start should not call initialize again', t => {
+  if (typeof document === 'undefined') {
+    t.comment('browser-only test');
+    t.end();
+    return;
+  }
+
+  const {gl} = fixture;
+  let initializeCalled = 0;
+
+  const animationLoop = new AnimationLoop({
+    gl,
+    onInitialize: () => {
+      initializeCalled++;
+    }
+  });
+  animationLoop.start();
+  setTimeout(() => animationLoop.stop(), 50);
+  setTimeout(() => animationLoop.start(), 100);
+  setTimeout(() => {
+    t.is(initializeCalled, 1, 'onInitialize called');
+    t.end();
+  }, 150);
```

Comment:
These `setTimeout`s are a bit ugly. I'm just trying to get past the async `getPageLoadPromise()` [here](https://github.com/uber/luma.gl/blob/master/modules/core/src/core/animation-loop.js#L105).